src/standalone/Vegetation/photosynthesis.jl

```diff
@@ -1,386 +1,298 @@
+and Aj prior to this function.
+
+See Table 11.5 of G. Bonan's textbook, Climate Change and Terrestrial Ecosystem Modeling (2019).
+"""
+function gross_photosynthesis(Ac::FT, Aj::FT) where {FT}
+    return min(Ac, Aj)
 end
 
-function FarquharModel{FT}(
-    parameters::FarquharParameters{FT},
-) where {FT <: AbstractFloat}
-    return FarquharModel{eltype(parameters), typeof(parameters)}(parameters)
+"""
+    rubisco_assimilation(is_c3::AbstractFloat, args...)
+
+Calls the correct rubisco assimilation function based on the `is_c3`.
+
+A `is_c3` value of 1.0 corresponds to C3 photosynthesis and calls
+`c3_rubisco_assimilation`, while 0.0 corresponds to C4 photsynthesis and calls
+`c4_rubisco_assimilation`.
+"""
+function rubisco_assimilation(is_c3::AbstractFloat, args...)
+    is_c3 > 0.5 ? c3_rubisco_assimilation(args...) :
+    c4_rubisco_assimilation(args...)
 end
 
-ClimaLand.name(model::AbstractPhotosynthesisModel) = :photosynthesis
-ClimaLand.auxiliary_vars(model::FarquharModel) = (:An, :GPP, :Rd)
-ClimaLand.auxiliary_types(model::FarquharModel{FT}) where {FT} = (FT, FT, FT)
-ClimaLand.auxiliary_domain_names(::FarquharModel) =
-    (:surface, :surface, :surface)
-
-
-function photosynthesis_at_a_point_Farquhar(
-    T,
-    β,
-    Rd,
-    APAR,
-    c_co2,
-    medlyn_factor,
-    R,
-    Vcmax25,
-    is_c3,
-    Γstar25,
-    ΔHJmax,
-    ΔHVcmax,
-    ΔHΓstar,
-    fC3,
-    fC4,
-    ΔHRd,
-    To,
-    θj,
-    ϕ,
-    oi,
-    Kc25,
-    Ko25,
-    ΔHkc,
-    ΔHko,
-    Q10,
-    s1,
-    s2,
-    s3,
-    s4,
-    s5,
-    s6,
-    E,
-)
-    Jmax = max_electron_transport(Vcmax25, ΔHJmax, T, To, R)
-    J = electron_transport(APAR, Jmax, θj, ϕ)
-    Vcmax =
-        compute_Vcmax(is_c3, Vcmax25, T, R, To, ΔHVcmax, Q10, s1, s2, s3, s4)
-    Γstar = co2_compensation(Γstar25, ΔHΓstar, T, To, R)
-    ci = intercellular_co2(c_co2, Γstar, medlyn_factor)
-    Aj = light_assimilation(is_c3, J, ci, Γstar, APAR, E)
-    Kc = MM_Kc(Kc25, ΔHkc, T, To, R)
-    Ko = MM_Ko(Ko25, ΔHko, T, To, R)
-    Ac = rubisco_assimilation(is_c3, Vcmax, ci, Γstar, Kc, Ko, oi)
-    return net_photosynthesis(Ac, Aj, Rd, β)
+"""
+    c3_rubisco_assimilation(Vcmax::FT,
+                         ci::FT,
+                         Γstar::FT,
+                         Kmm::FT) where {FT}
+
+Computes the Rubisco limiting rate of photosynthesis for C3 plants (`Ac`),
+in units of moles CO2/m^2/s,
+as a function of the maximum rate of carboxylation of Rubisco (`Vcmax`),
+the leaf internal carbon dioxide partial pressure (`ci`),
+the CO2 compensation point (`Γstar`), and the effective Michaelis-Menten parameter
+`K_mm`.
+
+The units of ci, Γstar, and Kmm may be Pa or mol/mol, but they must be consistent. The
+units of Vcmax must be mol CO2/m^2.s.
+
+See Table 11.5 of G. Bonan's textbook, Climate Change and Terrestrial Ecosystem Modeling (2019).
+"""
+function c3_rubisco_assimilation(
```

Comment:
I made it to farquhar computes Kmm and then it uses this function as well, instead of having a separate function for pmodel and farquhar (where farquhar computed Kmm internally, and the pmodel function took it as input)